pkg/ipcache/kvstore.go

```diff
@@ -312,15 +312,6 @@ func (iw *IPIdentityWatcher) OnUpdate(k storepkg.Key) {
 	}
 
 	peerIdentity := ipIDPair.ID
-	if option.Config.EnableRemoteNodeIdentity && peerIdentity == identity.ReservedIdentityHost {
```

Comment:
This should still compare `peerIdentity` to `identity.ReservedIdentityHost` and set `peerIdentity` to `ReservedIdentityRemoteNode` if they are equal.
```suggestion
if peerIdentity == identity.ReservedIdentityHost {
```